src/game.ts

```diff
@@ -1,13 +1,159 @@
+      let currentDot: Point =
+        Math.floor(Math.random() * 2) == 0 ? flow.start : flow.end;
+
+      let direction: Direction = directions.at(Math.floor(Math.random() * 4))!;
+      let newDot: Point = {
+        row: currentDot.row + direction.at(0)!,
+        column: currentDot.column + direction.at(1)!,
+      };
+      if (isInvalidNewDot(newDot, this.boardSize)) continue;
+
+      // If newDot is on an existing dot
+      if (isOnAnotherDot(this.flows, flow, newDot)) {
+        if (pointsAreEqual(flow.start, currentDot)) {
+          flow.start = newDot;
+          flow.solution.unshift(newDot);
+        } else {
+          flow.end = newDot;
+          flow.solution.push(newDot);
+        }
+        i += 1;
+      }
+    }
+
+    // Finds the corners in the newly created lines (flow.solution)
+    this.flows.forEach((flow: Flow) => {
+      let solutionLine: Point[] = flow.solution;
+      flow.solution = [];
+
+      flow.solution.push(flow.start);
+
+      for (let i = 1; i < solutionLine.length - 1; i++)
+        if (
+          solutionLine.at(i - 1)!.row != solutionLine.at(i + 1)!.row &&
+          solutionLine.at(i - 1)!.column != solutionLine.at(i + 1)!.column
+        )
+          flow.solution.push(solutionLine.at(i)!);
+
+      flow.solution.push(flow.end);
+    });
+
+    this.timeStart = new Date();
+  }
+  public initializeLevel(): void {
+    Object.values(Color)
+      .slice(0, this.boardSize)
+      .sort(() => Math.floor(Math.random() * 3) - 1)
+      .forEach((_color: string | Color, i: number) => {
+        let solution: Point[] = [];
+        for (let row = 0; row < this.boardSize; row++)
+          solution.push({ row: row, column: i });
+        this.flows.push(
+          new Flow(
+            i,
+            { row: 0, column: i },
+            { row: this.boardSize - 1, column: i },
+            solution
+          )
+        );
+      });
+  }
+
+  public toggleSolution(): void {
+    this.showSolution = !this.showSolution;
+  }
+
+  public caseClicked(pos: Point) {
+    if (!this.showSolution && !this.isBoardCompleted()) {
+      this.flows.forEach((flow: Flow) => {
+        if (pointsAreEqual(flow.start, pos) || pointsAreEqual(flow.end, pos!)) {
+          flow.lines = [pos];
+          flow.corners = [pos];
+          flow.completed = false;
+          this.currentFlow = flow;
+        }
+      });
+    }
+  }
 
-  constructor(boardSize: number) {
-    this.boardSize = boardSize;
-    this.createGame();
+  public isBoardCompleted(): boolean {
+    if (!this.flows.every((flow: Flow) => flow.completed)) return false;
+
+    let totalLines = 0;
+    this.flows.forEach((flow: { lines: string | any[] }) => {
+      totalLines += flow.lines.length;
+    });
+    return totalLines == this.boardSize * this.boardSize;
+  }
+
+  private resetLine(): void {
+    if (!this.currentFlow) return;
+    this.currentFlow.lines = [];
+    this.currentFlow.corners = [];
+    this.currentFlow = undefined;
```

Comment:
Si tu set `currentFlow` à `undefined`, tu n'as pas besoin de setter ses propriétés aux deux lignes précédentes.

Reply:
no I need it